src/util/map.js

```diff
@@ -122,3 +122,26 @@ export function mapEntries<K, V, InK, InV>(
   }
   return result;
 }
+
+/**
+ * Concatenate maps without mutating the arguments.
+ *
+ * Combines multiple maps, returning a new map which has every key from
+ * the source maps, with their corresponding values. None of the inputs
+ * are mutated. In the event that multiple maps have the same key, an
+ * error will be thrown.
+ */
+export function concat<K, V>(maps: $ReadOnlyArray<Map<K, V>>): Map<K, V> {
```

Comment:
It would be nice for test cases like this to work:

```js
const numberMap: Map<number, number> = new Map().set(1, 2);
const stringMap: Map<string, string> = new Map().set("one", "two");
type NS = number | string;
const polyMap: Map<NS, NS> = concat([numberMap, stringMap]);
// (result could also be `Map<mixed, mixed>`)
```

Or, perhaps more usefully:

```js
const gitNodeDescriptions: Map<GitAddress, string> = new Map();
const githubNodeDescriptions: Map<GithubAddress, string> = new Map();
const nodeDescriptions: Map<NodeAddressT, string> = concat([
  gitNodeDescriptions,
  githubNodeDescriptions,
]);
```

I believe (but have not tested) that (a) this test case would not pass
currently, and (b) it would pass if you changed the type declaration to

```js
concat<K, V>(maps: $ReadOnlyArray<Map<$Subtype<K>, $Subtype<V>>>): Map<K, V>;
```

What do you think of this suggestion?

Reply:
Done, thanks.